app/models/app_remote_kill.rb

```diff
@@ -0,0 +1,77 @@
+class AppRemoteKill < ActiveRecord::Base
+  belongs_to :app
+  belongs_to :split
+
+  attribute :first_bad_version, :app_version
+  attribute :fixed_version, :app_version
+
+  validates :app, :split, :reason, :override_to, :first_bad_version, presence: true
+  validates :reason, uniqueness: { scope: %i(app split) }
+
+  validate :override_to_must_exist
+  validate :fixed_version_must_be_greater_than_first_bad_version
+  validate :must_not_overlap_existing
+
+  scope :affecting, ->(app_build, override: false, overridden_at: nil) do
```

Comment:
Taking a stab at what this basically translates to in SQL:

```
WHERE app_id = {{app_id}}
  AND first_bad_version <= {{version}}
  AND fixed_version IS NULL OR fixed_version > {{version}}
  AND {{"assignments.force" or true/false}} = false OR updated_at > {{"assignments.updated_at" or NULL}}
```

I'm wondering if you could save the query engine some work if you get non-Arel nodes for arguments. If `override` is true and `overridden_at` is `nil`, you can essentially change this whole thing to `WHERE false = true`, no? Maybe not worth it?

Reply:
override = true and overriden_at is null is actually an illegal state, you either get both or neither. I needed them to be expressable as distinct arel nodes, though.

Also this is coming out in #112

Reply:
also the query engine will resolve literal-based logic to `WHERE false = true` before it does a single index scan, so i don't think we'd be interested in doing that logic in ruby instead of PG anyway